includes/common.inc

```diff
@@ -3834,6 +3834,7 @@ function _drupal_load_stylesheet($matches) {
  */
 function drupal_clear_css_cache() {
   variable_del('drupal_css_cache_files');
+  drupal_aggregated_file_flush('css');
```

Comment:
Flushing the cache for file existence should happen only after the deletions. Otherwise, it creates a race condition where the cache gets repopulated in a concurrent request with files that will end up deleted.